gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -80,11 +82,20 @@
   // instead we must retain explicit reference to the concrete type.
   private final PublishSubject<MapBoxOfflineTileProvider> tileProviders = PublishSubject.create();
 
+  /**
+   * Manager for handling click events for markers.
+   *
+   * <p>This isn't needed if the map only has a single layer. But in our case, multiple GeoJSON
+   * layers might be added and we wish to have independent clickable features for each layer.
+   */
+  private final MarkerManager markerManager;
+  // TODO: Add managers for polyline and polygon layers
+
   /**
    * References to Google Maps SDK Markers present on the map. Used to sync and update markers with
    * current view and data state.
    */
-  private Set<Marker> markers = new HashSet<>();
+  private final Collection markers;
```

Comment:
Could we qualify Collection with its outer class name (`MapObjectManager.Collection`)? I automatically assumed this was a java.util.Collection at first :)

Reply:
Done